### PR TITLE
Updating docs [SA-14580] (master)

### PR DESCRIPTION
**Pull request by tallzeebaa**
There still exists places to fix. I'll log them as separate tickets.
Mainly this PR focused on
* fixed issues found by lint 
* fixed issues pointed out by phpstorm plugin
* updated data type to match actual (not 100% covered)
* deleted ineffective properties listed with $ref

To see the difference of lint result, run: `npm test` 
(I have turned off `operation-4xx-response` check for now)

### CODE REVIEW

**Comment by jleckie** [comment]
You mention fixes found by lint.  Is this something we should be running on PRs as a github action or something to ensure the quality of our API docs?

**Comment by tallzeebaa** [comment]
Yes, we should run lint on every PR.

**Comment by armatronic** [comment]
> Yes, we should run lint (`npm test`) on every PR.

If that's the case: that should be added to the `npm run build` script in `package.json`. (We could then look at adding this to CI.)

**Review by armatronic — COMMENTED** [review]
Of course I'm a little worried about this PR given the number of changes it makes!

One thing I'd like to know is how do we verify if the changes we've made actually match what Schoolbox does? I'm supposing we don't have an easy way to test that?

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "private": true,
   "scripts": {
-    "build": "redocly bundle -o docs/builds/${npm_config_build_version}.yaml && ./scripts/addBuild.js ${npm_config_build_version} ${npm_config_selected} && ./scripts/applyVersion.js ${npm_config_build_version} docs/builds/${npm_config_build_version}.yaml",
+    "build": "redocly bundle -o docs/builds/${npm_config_build_version}.yaml && ./scripts/addBuild.js ${npm_config_build_version} ${npm_config_selected} && ./scripts/applyVersion.js ${npm_config_build_version} docs/builds/${npm_config_build_version}.yaml && redocly lint",
```
Comment (armatronic) [inline package.json:11]: Shouldn't `redocly lint` be run before the other commands? That way the build exits early if the spec is invalid.

File: openapi/components/schemas/pastoral-item.yaml [hunk 2]
```diff
@@ -1,124 +1,158 @@
...
+            and any other names, and in any case is not the definitive source of
+            naming information; therefore in practice this will usually be the
+            user's given name.
+          example: Rebecca
+        lastName:
+          type: string
+          nullable: true
+          description: The user's surname.
+          example: White
+        preferredName:
```
[the remark this answers is not in the document]
Comment (tallzeebaa) [reply openapi/components/schemas/pastoral-item.yaml:43]: > One thing I'd like to know is how do we verify if the changes we've made actually match what Schoolbox does? I'm supposing we don't have an easy way to test that?

Not in our product. In the previous job, the parameters were automatically generated from doc block and api methods are gathered in certain places, and thus there wasn't this problem.

[the remark this answers is not in the document]
Comment (armatronic) [reply openapi/components/schemas/pastoral-item.yaml:43]: > In the previous job, the api documents were automatically generated from doc block and api methods are gathered in certain places, and thus there wasn't this problem.

Are you suggesting something like https://zircote.github.io/swagger-php/guide/annotations.html?

[the remark this answers is not in the document]
Comment (tallzeebaa) [reply openapi/components/schemas/pastoral-item.yaml:43]: like this: https://www.phpdoc.org/

**Comment by armatronic** [comment]
In it goes :open_mouth: 

I'll create another PR for my communication around the changes.

**Pull request merged**